backend/src/v2/compiler/argocompiler/container_test.go

```diff
@@ -15,13 +15,107 @@
 package argocompiler
 
 import (
+	"os"
 	"testing"
 
 	wfapi "github.com/argoproj/argo-workflows/v3/pkg/apis/workflow/v1alpha1"
 	"github.com/kubeflow/pipelines/kubernetes_platform/go/kubernetesplatform"
 	"github.com/stretchr/testify/assert"
 )
 
+func TestAddContainerExecutorTemplate(t *testing.T) {
+	tests := []struct {
+		name                  string
+		configMapName         string
+		configMapKey          string
+		mountPath             string
+		expectedVolumeName    string
+		expectedConfigMapName string
+		expectedMountPath     string
+	}{
+		{
+			name:                  "Test with valid settings",
+			configMapName:         "kube-root-ca.crt",
+			configMapKey:          "ca.crt",
+			mountPath:             "/etc/ssl/custom",
+			expectedVolumeName:    "ca-bundle",
+			expectedConfigMapName: "kube-root-ca.crt",
+			expectedMountPath:     "/etc/ssl/custom/ca.crt",
+		},
+		// You can add more test cases here
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			// Setup the environment variables for testing
+			os.Setenv("EXECUTOR_CABUNDLE_CONFIGMAP_NAME", tt.configMapName)
+			os.Setenv("EXECUTOR_CABUNDLE_CONFIGMAP_KEY", tt.configMapKey)
+			os.Setenv("EXECUTOR_CABUNDLE_MOUNTPATH", tt.mountPath)
+
+			t.Log("executor configmap", os.Getenv("EXECUTOR_CABUNDLE_CONFIGMAP_NAME"))
+
+			defer func() {
+				// Clean up environment variables
+				os.Unsetenv("EXECUTOR_CABUNDLE_CONFIGMAP_NAME")
+				os.Unsetenv("EXECUTOR_CABUNDLE_CONFIGMAP_KEY")
+				os.Unsetenv("EXECUTOR_CABUNDLE_MOUNTPATH")
+			}()
```

Comment:
you probably don't need to call this every iteration, and can just call it at the end of the function

Reply:
done, updated to have the defer func in the end of the function.